web_ui/packages/ui/src/loading/loading.component.test.tsx

```diff
@@ -0,0 +1,328 @@
+            expect(progressbar).toBeInTheDocument();
+            expect(progressbar).toHaveAttribute('aria-label', 'Loading...');
+
+            // In inline mode, there should be no container with position absolute
+            const container = progressbar.closest('[style*="position: absolute"]');
+            expect(container).not.toBeInTheDocument();
+        });
+
+        it('renders with custom size', () => {
+            renderLoading({ mode: 'inline', size: 'S' });
+
+            const progressbar = screen.getByRole('progressbar');
+            expect(progressbar).toBeInTheDocument();
+        });
+
+        it('passes through custom props to ProgressCircle', () => {
+            renderLoading({
+                mode: 'inline',
+                'aria-label': 'Custom loading message',
+                value: 50,
+            });
+
+            const progressbar = screen.getByRole('progressbar');
+            expect(progressbar).toHaveAttribute('aria-label', 'Custom loading message');
+        });
+    });
+
+    describe('fullscreen mode', () => {
+        it('renders fullscreen loading with overlay by default', () => {
+            renderLoading();
+
+            const progressbar = screen.getByRole('progressbar');
+            expect(progressbar).toBeInTheDocument();
+            expect(progressbar).toHaveAttribute('aria-label', 'Loading...');
+
+            // Should render overlay container with CSS classes
+            const container = progressbar.parentElement?.parentElement;
+            expect(container).toBeInTheDocument();
+            expect(container).toHaveClass('overlay', 'fullscreen');
+        });
+
+        it('renders with custom id and testid', () => {
+            renderLoading({ id: 'custom-loading-id' });
+
+            const container = screen.getByTestId('custom-loading-id');
+            expect(container).toBeInTheDocument();
+            expect(container).toHaveAttribute('id', 'custom-loading-id');
+        });
+
+        it('applies default background color for fullscreen mode', () => {
+            renderLoading({ mode: 'fullscreen' });
+
+            const progressbar = screen.getByRole('progressbar');
+            // Find the container by traversing up the DOM tree
+            const container = progressbar.parentElement?.parentElement;
+            expect(container).toHaveStyle({ backgroundColor: 'var(--spectrum-global-color-gray-50)' });
+        });
+
+        it('applies custom background color', () => {
+            renderLoading({
+                mode: 'fullscreen',
+                style: { backgroundColor: 'rgba(255, 0, 0, 0.5)' },
+            });
+
+            const progressbar = screen.getByRole('progressbar');
+            const container = progressbar.parentElement?.parentElement;
+            expect(container).toHaveStyle({
+                backgroundColor: 'rgba(255, 0, 0, 0.5)',
+            });
+        });
+
+        it('applies positioning props', () => {
```

Comment:
could we not merge all of these tests into 1? Just to check if the style props are being passed? Which is already a bit overkill but never too bad to have.